endpoints/openrtb2/auction.go

```diff
@@ -1467,6 +1439,29 @@ func fillChannel(reqWrapper *openrtb_ext.RequestWrapper, isAmp bool) error {
 
 }
 
+func (deps *endpointDeps) validateSourceSChain(req *openrtb_ext.RequestWrapper) error {
+	if req.Source == nil || req.Source.SChain == nil {
+		return nil
+	}
+	sChain := req.Source.SChain
+
+	if len(sChain.Ver) == 0 {
+		return errors.New("request.source.schain.ver cannot be empty")
+	}
+	if len(sChain.Nodes) == 0 {
+		return errors.New("request.source.schain.nodes cannot be empty")
+	}
+	for i, node := range sChain.Nodes {
+		if len(node.ASI) == 0 {
+			return fmt.Errorf("request.source.schain.nodes[%d].asi cannot be empty", i)
+		}
+		if len(node.SID) == 0 {
+			return fmt.Errorf("request.source.schain.nodes[%d].sid cannot be empty", i)
+		}
+	}
```

Comment:
I found this in [OpenRTB 2.6 spec](https://iabtechlab.com/wp-content/uploads/2022/04/OpenRTB-2-6_FINAL.pdf), paragraph 3.2.25 and 3.2.26.
 
You are probably referring to this required field: `Complete int8`. This field bus not a pointer and will always present in this struct. Specification also says it can `0 = no, 1 = yes`. Do we want to validate this value is 0 or 1 only? I missed this part.

Reply:
Let's remove this validation. PBS's approach is to just validate the schema (attributes & datatypes) for fields we do not use.

Reply:
Reverted